blueoil/cmd/predict.py

```diff
@@ -13,21 +13,160 @@
+from blueoil.utils.image import load_image
+from blueoil.utils import config as config_util
+from blueoil.utils.executor import search_restore_filename
+from blueoil.utils.predict_output.writer import OutputWriter
 
-def predict(input, output, experiment_id, checkpoint=None, save_images=True):
-    """Predict input images."""
+DUMMY_FILENAME = "DUMMY_FILE"
 
-    output_dir = os.environ.get("OUTPUT_DIR", "saved")
 
-    if checkpoint is None:
-        restore_path = None
-    else:
-        restore_path = os.path.join(
-            output_dir, experiment_id, "checkpoints", checkpoint
+def _get_images(filenames, pre_processor, data_format):
+    """ """
+    images = []
+    raw_images = []
+
+    for filename in filenames:
+        if filename == DUMMY_FILENAME:
+            raw_image = np.zeros((64, 64, 3), dtype=np.uint8)
+        else:
+            raw_image = load_image(filename)
+
+        image = pre_processor(image=raw_image)['image']
+        if data_format == 'NCHW':
+            image = np.transpose(image, [2, 0, 1])
+
+        images.append(image)
+        raw_images.append(raw_image)
+
+    return np.array(images), np.array(raw_images)
+
+
+def _all_image_files(directory):
+    all_image_files = []
+    for file_path in glob(os.path.join(directory, "*")):
+        if os.path.isfile(file_path) and imghdr.what(file_path) in {"jpeg", "png"}:
+            all_image_files.append(os.path.abspath(file_path))
+
+    return all_image_files
+
+
+def _run(input_dir, output_dir, config, restore_path, save_images):
+    ModelClass = config.NETWORK_CLASS
+    network_kwargs = dict((key.lower(), val) for key, val in config.NETWORK.items())
+
+    graph = tf.Graph()
+    with graph.as_default():
+        model = ModelClass(
+            classes=config.CLASSES,
+            is_debug=config.IS_DEBUG,
+            **network_kwargs
         )
 
-    run_predict(input, output, experiment_id, None, restore_path, save_images)
+        is_training = tf.constant(False, name="is_training")
+
+        images_placeholder, _ = model.placeholders()
+        output_op = model.inference(images_placeholder, is_training)
+
+        init_op = tf.global_variables_initializer()
+
+        saver = tf.compat.v1.train.Saver(max_to_keep=None)
+
+    session_config = tf.ConfigProto()
+    sess = tf.Session(graph=graph, config=session_config)
+    sess.run(init_op)
+    saver.restore(sess, restore_path)
+
+    all_image_files = _all_image_files(input_dir)
+
+    step_size = int(math.ceil(len(all_image_files) / config.BATCH_SIZE))
+
+    writer = OutputWriter(
+        task=config.TASK,
+        classes=config.CLASSES,
+        image_size=config.IMAGE_SIZE,
+        data_format=config.DATA_FORMAT
+    )
+
+    results = []
+    for step in range(step_size):
+        start_index = (step) * config.BATCH_SIZE
+        end_index = (step + 1) * config.BATCH_SIZE
+
+        image_files = all_image_files[start_index:end_index]
+
+        while len(image_files) != config.BATCH_SIZE:
```

Comment:
this is inefficient, and will loop forever if len(image_files) > config.BATCH_SIZE.
```
if len > size:
    files += [DUMMY] * (len - size)
```